src/main/java/io/github/darealturtywurty/ancientology/common/items/ForbiddenFruitItem.java

```diff
@@ -0,0 +1,151 @@
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+import net.minecraft.network.chat.TranslatableComponent;
+import net.minecraft.server.level.ServerLevel;
+import net.minecraft.server.level.ServerPlayer;
+import net.minecraft.world.effect.MobEffect;
+import net.minecraft.world.effect.MobEffectInstance;
+import net.minecraft.world.entity.EntityType;
+import net.minecraft.world.entity.LightningBolt;
+import net.minecraft.world.entity.LivingEntity;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.entity.projectile.Arrow;
+import net.minecraft.world.food.FoodProperties;
+import net.minecraft.world.item.Item;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.item.enchantment.Enchantment;
+import net.minecraft.world.level.Level;
+
+import io.github.darealturtywurty.ancientology.core.init.ItemInit;
+import net.minecraftforge.registries.ForgeRegistries;
+
+public class ForbiddenFruitItem extends Item {
+    public static final FoodProperties FOOD = new FoodProperties.Builder().alwaysEat().nutrition(10).saturationMod(0.1f).build();
+
+    public ForbiddenFruitItem(Properties pProperties) {
+        super(pProperties);
+    }
+
+    @Override
+    public ItemStack finishUsingItem(ItemStack pStack, Level pLevel, LivingEntity pLivingEntity) {
+        if (!pLevel.isClientSide)
+            useItem(pLevel, pLivingEntity);
+        return super.finishUsingItem(pStack, pLevel, pLivingEntity);
+    }
+
+    private void useItem(Level level, LivingEntity entity) {
+        double d = Math.random();
+        double f = Math.random();
+        if (d < 0.99) {
+            if (f < 0.1) {
+                sendToNether(entity);
+            } else if (f < 0.2) {
+                lightningStorm(entity, level);
+            } else if (f < 0.25) {
+                removeHunger(entity);
+            } else if (f < 0.4) {
+                giveRandomBadEffect(entity);
+            } else if (f < 0.5) {
+                rainOfArrows(entity, level);
+            } else if (f < 0.52) {
+                secondChance(entity);
+            } else if (f < 0.6) {
+                //TODO
+            }
+        } else {
+            if (f < 0.7) {
+                giveRandomItem(entity);
+            } else {
+                giveRandomGoodEffect(entity);
+            }
+        }
+    }
+
+    public void secondChance(LivingEntity entity) {
+        entity.sendMessage(new TranslatableComponent("msg.acientology.fruit_chance"), entity.getUUID());
+        if (entity instanceof Player player) {
+            player.addItem(new ItemStack(ItemInit.FORBIDDEN_FRUIT.get()));
+        }
+    }
+
+    public void rainOfArrows(LivingEntity entity, Level level) {
+        entity.sendMessage(new TranslatableComponent("msg.acientology.fruit_arrow_rain"), entity.getUUID());
+        for (int j = 0; j < 30; j++) {
+            Arrow bolt = new Arrow(EntityType.ARROW, level);
```

Comment:
`var` can be used here.

Reply:
Why though, is it more efficient? It's certainly not more readable.

Reply:
It is in-fact more readable. Yes.